db/functions/base.py

```diff
@@ -306,6 +307,19 @@ class ToLowercase(DBFunction):
     def to_sa_expression(string):
         return func.lower(string)
 
+class ArrayLength(DBFunction):
+    id = 'array_length'
+    name = 'length'
+    hints = tuple([
+        hints.returns(hints.comparable),
+        hints.parameter_count(1),
+        hints.parameter(0, hints.json_array),
+        hints.mathesar_filter,
```

Comment:
This isn't actually a filter. You'll probably use this function to compose a `DBFunctionPacked`-based filter and you'll give that a `mathesar_filter` hint, but giving this function here a `mathesar_filter` will result in unexpected behaviour.

Reply:
Thanks for your very helpful explanation! @dmos62 , I added a DBFunctionPacked function using ArrayLength in the `db/types/custom/json_array` and also moved the ArrayLength into that file. Also, I added the json array into categories.